File: backend/DownloadManager.Shared/Models/Category.cs
namespace DownloadManager.Shared.Models;

public class Category
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string FolderPath { get; set; } = string.Empty;
    public string? FileExtensions { get; set; }
    public string? MimeTypes { get; set; }
    public bool IsSystem { get; set; }
    public string? Color { get; set; }
    public string? Icon { get; set; }
}

File: ui/src/types/category.ts
export interface Category {
  id: number;
  name: string;
  folderPath: string;
  fileExtensions?: string;
  mimeTypes?: string;
  isSystem: boolean;
  color?: string;
  icon?: string;
}

export interface CategoryFormData {
  name: string;
  folderPath: string;
  fileExtensions: string[];
  mimeTypes: string[];
  color: string;
  icon: string;
}
